src/test/java/core/basesyntax/service/impl/ReportServiceImplTest.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax.service.impl;
+
+import static core.basesyntax.model.Product.APPLE;
+import static core.basesyntax.model.Product.BANANA;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportService;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class ReportServiceImplTest {
+    private static ReportService reportService;
+
+    @BeforeAll
+    static void beforeAll() {
+        reportService = new ReportServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.STORAGE_MAP.clear();
+    }
+
+    @DisplayName("Check report with banana and apple")
+    @Test
+    void createReport_bananaAndApple_ok() {
+        Storage.STORAGE_MAP.put(BANANA.getName(), 152);
+        Storage.STORAGE_MAP.put(APPLE.getName(), 90);
+        List<String> expected = List.of("banana,152", "apple,90").stream()
+                .map(String::trim).collect(Collectors.toList());
+        List<String> actual = reportService.createReport().stream()
+                .map(String::trim).collect(Collectors.toList());
+        assertEquals(expected, actual);
+    }
+
+    @DisplayName("Check report with banana")
+    @Test
+    void createReport_banana_ok() {
+        Storage.STORAGE_MAP.put(BANANA.getName(), 99);
+        List<String> expected = List.of("banana,99").stream()
+                .map(String::trim).collect(Collectors.toList());
+        List<String> actual = reportService.createReport().stream()
+                .map(String::trim).collect(Collectors.toList());
```

Comment:
this code can be written in more concise way:
```suggestion
        List<String> expected = List.of("banana,99");
        List<String> actual = reportService.createReport();
```
please check and fix other places if needed.

Reply:
I would be happy to write it in a simpler way, however, this way the test will not pass, so I have to leave it :(